client/src/lib.rs

```diff
@@ -74,28 +90,141 @@ pub enum SubxtClientError {
+                                    }));
+
+                                let (mut send_front_sub, send_back_sub) =
+                                    mpsc::channel(4);
+                                if let Ok(message) = serde_json::to_string(&request) {
+                                    if let Some(response) =
+                                        rpc.rpc_query(&session, &message).await
+                                    {
+                                        let result = match serde_json::from_str::<Output>(
+                                            &response,
+                                        )
+                                        .expect("failed to decode subscription response")
+                                        {
+                                            Output::Success(_) => {
+                                                Ok((
+                                                    send_back_sub,
+                                                    SubscriptionId::Num(0),
+                                                ))
+                                            }
+                                            Output::Failure(failure) => {
+                                                Err(JsonRpseeError::Request(
+                                                    failure.error,
+                                                ))
+                                            }
+                                        };
+
+                                        send_back.send(result).expect(
+                                            "failed to send subscription response",
+                                        );
+                                    }
+                                }
+
+                                task::spawn(async move {
+                                    let mut from_back = from_back.compat();
+                                    let _session = session.clone();
+
+                                    while let Some(Ok(response)) = from_back.next().await
+                                    {
+                                        let notif = serde_json::from_str::<
+                                            SubscriptionNotif,
+                                        >(
+                                            &response
+                                        )
+                                        .expect("failed to decode subscription notif");
+                                        send_front_sub
+                                            .send(notif.params.result)
+                                            .await
+                                            .expect("failed to send subscription notif")
+                                    }
+                                });
+                            }
+
+                            FrontToBack::SubscriptionClosed(_) => {
+                                // TODO: implement
```

Comment:
it's fine the skip this but it might load on the network because the server will never be notified that we are ignoring the messages.

Reply:
agreed, should be minimal since the embedded client is destroyed at the end of each test but I'll note to re-write this later